supervisor/supervisor.go

```diff
@@ -275,8 +277,12 @@ func (s *Supervisor) Start() error {
 		"cpus":        s.machine.Cpus,
 	}).Debug("containerd: supervisor running")
 	go func() {
-		for i := range s.tasks {
-			s.handleTask(i)
+		for {
+			select {
+			case i := <-s.tasks:
```

Comment:
all the tasks should be (made?) context aware I suppose. But in the case that the client doesn't put a timeout on its context the issue remain.

We really want the operations to be serialized per containers though, so maybe break down into sub channel per container?